api/src/main/java/com/epam/pipeline/manager/user/UserImportManager.java

```diff
@@ -0,0 +1,241 @@
+import com.epam.pipeline.manager.metadata.MetadataManager;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.collections4.ListUtils;
+import org.apache.commons.collections4.MapUtils;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class UserImportManager {
+    private final UserManager userManager;
+    private final CategoricalAttributeManager categoricalAttributeManager;
+    private final MetadataManager metadataManager;
+    private final RoleManager roleManager;
+
+    /**
+     * Registers a new {@link PipelineUser}, {@link Role}, {@link MetadataEntry} for users
+     * and {@link CategoricalAttribute} if allowed. Otherwise, log event and skip action.
+     * @param createUser true if user shall be created if not exists
+     * @param createGroup true if role shall be created if not exists
+     * @param attributesToCreate the list of metadata keys that shall be created if not exists
+     * @param file the input file with users
+     * @return the list of events that happened during user processing
+     */
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    public List<PipelineUserEvent> importUsersFromFile(final boolean createUser, final boolean createGroup,
+                                                       final List<String> attributesToCreate,
+                                                       final MultipartFile file) {
+        final List<PipelineUserEvent> events = new ArrayList<>();
+        final List<CategoricalAttribute> categoricalAttributes = ListUtils
+                .emptyIfNull(categoricalAttributeManager.loadAll());
+        final List<PipelineUserWithStoragePath> users =
+                new UserImporter(events, categoricalAttributes, attributesToCreate).importUsers(file);
+        categoricalAttributeManager.updateCategoricalAttributes(categoricalAttributes);
+
+        users.forEach(user -> {
+            try {
+                events.addAll(ListUtils.emptyIfNull(processUser(user, createUser, createGroup, categoricalAttributes)));
+            } catch (Exception e) {
+                log.error(String.format("Failed to process user '%s'", user.getUserName()), e);
+                events.add(PipelineUserEvent.error(user.getUserName(), e.getMessage()));
+            }
+        });
+        return events;
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED)
```

Comment:
I don't think that `@Transactional` will work in this case as it is called from public non-transcation methods in the same class